packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -133,18 +145,19 @@ class MouseTracker extends ChangeNotifier {
     _trackedAnnotations.remove(annotation);
   }
 
-  bool _postFrameCheckScheduled = false;
+  bool _scheduledPostFramePositionCheck = false;
+  // Schedules a position check at the end of this frame for those annotations
+  // that have been added.
   void _scheduleMousePositionCheck() {
     // If we're not tracking anything, then there is no point in registering a
     // frame callback or scheduling a frame. By definition there are no active
     // annotations that need exiting, either.
-    if (_trackedAnnotations.isNotEmpty && !_postFrameCheckScheduled) {
-      _postFrameCheckScheduled = true;
-      SchedulerBinding.instance.addPostFrameCallback((Duration _) {
-        _postFrameCheckScheduled = false;
-        collectMousePositions();
+    if (_trackedAnnotations.isNotEmpty && !_scheduledPostFramePositionCheck) {
+      _scheduledPostFramePositionCheck = true;
+      SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+        sendMouseNotifications(_lastMouseEvent.keys.toSet());
```

Comment:
nit: we might not need `toSet` here since `keys` should have guaranteed uniqueness

Reply:
Good point. Removed.